src/Aspire.Dashboard/Protos/Partials.cs

```diff
@@ -49,5 +50,15 @@ ImmutableArray<EndpointViewModel> GetEndpoints()
                 .Select(e => new EndpointViewModel(e.EndpointUrl, e.ProxyUrl))
                 .ToImmutableArray();
         }
+
+        static T ValidateNotNull<T>(T value, [CallerArgumentExpression(nameof(value))] string? expression = null) where T : class
+        {
+            if (value is null)
+            {
+                throw new InvalidOperationException($"Message field '{expression}' cannot be null.");
+            }
+
+            return value;
```

Comment:
super nit:
```suggestion
            return value is null
                    ? throw new InvalidOperationException($"Message field '{expression}' cannot be null.")
                    : value;
```

Reply:
Could also be `return value ?? throw new ...` but I think I prefer the `if` style thanks.